core/cache/application.go

```diff
@@ -53,6 +54,10 @@ func (a *Application) WatchConfig(keys ...string) *ConfigWatcher {
 func (a *Application) setDetails(details ApplicationChange) {
 	a.mu.Lock()
 
+	if a.details.CharmURL != details.CharmURL {
+		a.hub.Publish(a.modelTopic(applicationCharmURLChange), []string{a.details.Name, details.CharmURL})
```

Comment:
Can we use a type for this message? I will probably need to include a generational branch name in the data at a later date.

Reply:
sure.